### DISCUSSION TITLE
About localStorage and API cancelled

### DISCUSSION THREAD

**Discussion by qwenhn** [alephjs/aleph.js#515] [Q&A] [answered]
Hi all.
Today, I created new project with Deno, Aleph and React. But there are a few things i don't know, I'm looking for advice:
1. I see in the example using localstorage but i don't see it save to Console -> Application -> Storage -> LocalStorage
2. When i toggle index page and todos page then i see API todos/_data_ is Cancelled. What is the reason here?

Thank so much. Have a nice day!

**Comment by ije** [comment, marked as answer]
1. since deno supports `localStorage` it saves data in deno cache system at ssr
2. it's react 18 issues, the `useEffect` will be invoked twice in dev mode

**Reply by qwenhn** [reply]
thanks @ije
